app/renderer/js/pages/preference/general-section.ts

```diff
@@ -389,11 +390,78 @@ export default class GeneralSection extends BaseSection {
 		}
 	}
 
-	updateResetDataOption(): void {
-		const resetDataButton = document.querySelector('#resetdata-option .reset-data-button');
+	/* Reset App Data */
+	resetAppData(): void {
+		const resetAppButton = document.querySelector('#resetdata-option .reset-data-button');
+		resetAppButton.addEventListener('click', () => {
+			this.resetAppSettings();
+		});
+	}
+
+	async resetAppSettings(): Promise<void> {
+		const resetMessage = 'Selecting YES will reset all the settings to default without removing any connected organizations and preferences from Zulip.';
+		// We save the Application's settings/configurations in following files
+		// Preferences file is able to reset zoom levels but requires restart, not hard reload
+		const settingFiles = ['config/settings.json', 'config/window-state.json', 'Partitions/webviewsession/Preferences'];
```

Comment:
@timabbott @akashnimare any suggestions on what else can be done with `Reset App Data`?

Reply:
Can you show what this looks like? I don't understand why you have "Selecting YES will ..." in the label for this setting.  We don't have similar extra verbosity in other settings, and I just specifically suggested removing such extra verbosity yesterday.

Reply:
Earlier messages were: 
_By proceeding you will be removing all connected organizations and preferences from Zulip._
and
_By clicking proceed you will be removing all added accounts and preferences from Zulip. When the application restarts, it will be as if you are starting Zulip for the first time._

Should we keep a simple "Do you want to proceed?" or "Are you sure you want to proceed?"

Reply:
![Screenshot 2020-04-09 at 1 44 27 AM](https://user-images.githubusercontent.com/31139861/78829261-a7fba980-7a03-11ea-82a3-419978d30713.png)
This is how it, at present in this PR, looks like

I totally agree on the part: 
> I just specifically suggested removing such extra verbosity yesterday.
>
Just left this as we already displayed these messages

Reply:
@timabbott ping